Code/Framework/AzToolsFramework/AzToolsFramework/PropertyTreeEditor/PropertyTreeEditor.cpp

```diff
@@ -914,7 +914,14 @@ namespace AzToolsFramework
 
             bool addChildren = true;
             auto editMetaData = node.GetElementEditMetadata();
-            if (editMetaData)
+
+            if (editMetaData && !editMetaData->m_name)
+            {
+                AZStd::string nodeName = node.GetElementMetadata()->m_name;
+                AZ_Warning("PropertyTreeEditor", false, "Found ElementData with no name on node \"%s\" with path\"%s\", skipping.", nodeName.c_str(), path.c_str());
+            }
+
+            if (editMetaData && editMetaData->m_name)
```

Comment:
Could this be condensed?

```
    if (editMetaData)
    {
        if(!editMetaData->m_name)
        {
            // ...
        }
        else
        {
            // ....
```

Reply:
You could do it that way but I was intentionally avoiding the extra level of indentation.

Reply:
I'll condense the first if and just roll it into the warning